### PR TITLE
Relation list shouldn't link to remote apps

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #3151.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- connect to the status with some local/remote apps
- only the local ones should link to the inspector and have icons

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8033/

**Comment by frankban** [comment]
QA is ok.

File: jujugui/static/gui/src/app/components/status/status.js [hunk 1]
```diff
@@ -589,6 +589,28 @@ class Status extends React.Component {
...
+    @param name {String} An app name.
+    @returns {Object} The link element.
+  */
+  _generateRelationAppLink(name) {
+    const app = this.props.db.services.getById(name);
+    if (!app) {
+      // If the application is not in the DB it must be remote app so don't
+      // link to it.
+      return (<span>{name}</span>);
+
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/status/status.js]: extra line

File: jujugui/static/gui/src/app/components/status/status.js [hunk 2]
```diff
@@ -589,6 +589,28 @@ class Status extends React.Component {
...
 
   /**
+    Generate a link to an application from a relation.
+    @param name {String} An app name.
+    @returns {Object} The link element.
+  */
+  _generateRelationAppLink(name) {
+    const app = this.props.db.services.getById(name);
+    if (!app) {
+      // If the application is not in the DB it must be remote app so don't
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/status/status.js]: I don't think this is a reliable/future proof way of doing this as I could see a time when we'd want all applications in the database for easy querying of information. Was there not something unique with the id's that we could trigger instead?

Comment (hatched) [reply jujugui/static/gui/src/app/components/status/status.js]: Sorry you can disregard this comment, I forgot that we store the remoteApplications in the db in another object.

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8036/

**Pull request merged**